java/clients/src/test/java/sleeper/clients/deploy/jar/PublishJarsToRepoTest.java

```diff
@@ -0,0 +1,176 @@
+        assertThat(commandsThatRan)
+                .containsExactly(clientJarPipeline1, clientJarPipeline2, lambdaJarPipeline1, lambdaJarPipeline2);
+    }
+
+    @Test
+    public void shouldThrowIllegalArgumentExceptionWhenTooFewArguments() {
+        assertThatThrownBy(() -> PublishJarsToRepo.main(new String[]{"anyString", "anyString"}))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("Usage: <jars-dir> <repository url> <m2 settings server id>");
+    }
+
+    @Test
+    public void shouldThrowIllegalArgumentExceptionWhenTooManyArguments() {
+        assertThatThrownBy(() -> PublishJarsToRepo.main(new String[]{"anyString", "anyString", "anyString", "anyString"}))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("Usage: <jars-dir> <repository url> <m2 settings server id>");
+    }
+
+    @Test
+    public void shouldThrowNullPointerWhenPathOfJarsDirectoryIsNull() {
+        assertThatThrownBy(() -> genericBuilder.pathOfJarsDirectory(null).build())
+                .isInstanceOf(NullPointerException.class)
+                .hasMessage("File path must not be null");
+    }
+
+    @Test
+    public void shouldThrowNullPointerWhenRepositoryUrlIsNull() {
+        assertThatThrownBy(() -> genericBuilder.repoUrl(null).build())
+                .isInstanceOf(NullPointerException.class)
+                .hasMessage("Repository URL must not be null");
+    }
+
+    @Test
+    public void shouldThrowNullPointerWhenm2SettingsServerIdIsNull() {
+        assertThatThrownBy(() -> genericBuilder.m2SettingsServerId(null).build())
+                .isInstanceOf(NullPointerException.class)
+                .hasMessage("M2 settings server Id must not be null");
+    }
+
+    @Test
+    public void shouldThrowNullPointerWhenVerisonIsNull() {
+        assertThatThrownBy(() -> genericBuilder.version(null).build())
+                .isInstanceOf(NullPointerException.class)
+                .hasMessage("Version to publish must not be null");
+    }
+
+    @Test
+    public void shouldThrowNullPointerWhenCommandRunnerIsNull() {
+        assertThatThrownBy(() -> genericBuilder.commandRunner(null).build())
+                .isInstanceOf(NullPointerException.class)
+                .hasMessage("Command Runner must not be null");
+    }
+
+    @Test
+    public void shouldThrowNullPointerWhenClientJarsIsNull() {
+        assertThatThrownBy(() -> genericBuilder.clientJars(null).build())
+                .isInstanceOf(NullPointerException.class)
+                .hasMessage("Client jars must not be null");
+    }
+
+    @Test
+    public void shouldThrowNullPointerWhenLambdaJarsIsNull() {
+        assertThatThrownBy(() -> genericBuilder.lambdaJars(null).build())
+                .isInstanceOf(NullPointerException.class)
+                .hasMessage("Lambda jars must not be null");
+    }
+
+    private CommandPipeline generateCommandPipeline(String filename, String artifactId) {
```

Comment:
The name of this helper method could describe its purpose, e.g. mavenDeployFileCommand.